### PR TITLE
[REVIEW] adding Corr() function inside dataframe.py to compute the correlation matrix of a data frame  and adding examples in series.py

### PR DESCRIPTION
**Pull request by rnyak**

### CODE REVIEW

File: python/cudf/cudf/tests/test_stats.py [hunk 1]
```diff
@@ -372,3 +373,11 @@ def test_corr1d(data1, data2):
     got = gs1.corr(gs2)
     expected = ps1.corr(ps2)
     np.testing.assert_approx_equal(got, expected, significant=8)
+ 
+def test_df_corr():
+    gdf = cudf.datasets.randomdata(1000000, {str(x):float for x in range(50)})
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_stats.py]: Could you also reduce the size? The idea is for these tests to be small to run fast and make debugging as easy as possible.
```suggestion
    gdf = randomdata(100, {str(x):float for x in range(50)})
```

File: python/cudf/cudf/tests/test_stats.py [hunk 2]
```diff
@@ -372,3 +373,11 @@ def test_corr1d(data1, data2):
...
+    pdf= gdf.to_pandas()
+    got = gdf.corr()
+    expected = pdf.corr()
+    np.testing.assert_array_almost_equal(got.to_pandas(), expected)
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_stats.py]: You'll need to import this from the test utils, but this streamlines things
```suggestion
    assert_eq(got, expected)
```

File: python/cudf/cudf/tests/test_stats.py [hunk 3]
```diff
@@ -372,3 +373,11 @@ def test_corr1d(data1, data2):
     got = gs1.corr(gs2)
     expected = ps1.corr(ps2)
     np.testing.assert_approx_equal(got, expected, significant=8)
+ 
+def test_df_corr():
+    gdf = cudf.datasets.randomdata(1000000, {str(x):float for x in range(50)})
+    pdf= gdf.to_pandas()
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_stats.py]:
```suggestion
    pdf = gdf.to_pandas()
```

**Comment by rnyak** [comment]
@kkraus14 I have done the changes you requested for the unit tests. I also have just added an example in the `ioutils.py` for reading hex values as int. FYI.

File: python/cudf/cudf/utils/ioutils.py [hunk 4]
```diff
@@ -771,6 +771,33 @@
 0  123 2018-11-13T12:00:00.000 5451
 1  456 2018-11-14T12:35:01.000 5784
 2  789 2018-11-15T18:02:59.000 6117
+"""
+
+"""
```
Comment (kkraus14) [inline python/cudf/cudf/utils/ioutils.py]: We shouldn't be breaking this string here I believe, we also don't need to redefine the `Examples` section

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 
 ## Improvements
-
+- PR #4140 Add cudf series examples in docs and corr() method for dataframe in dataframe.py
```
Comment (kkraus14) [inline CHANGELOG.md]:
```suggestion

- PR #4140 Add cudf series examples in docs and corr() method for dataframe in dataframe.py
```

File: python/cudf/cudf/utils/ioutils.py [hunk 6]
```diff
@@ -772,6 +772,29 @@
...
+>>> filename = 'test.csv'
+>>> cdf = cudf.DataFrame()
+>>> cdf['hex_col'] = ['9512c20b']*10
+>>> cdf.to_csv(fname,index=False)
+
+Read the file with ``cudf.read_csv`` and use `hex64` as dtype.
+dtype: 'hex64', 'hex32' or 'hex` (alias for hex64) for hexadecimal parsing.
+
+>>> gdf = cudf.read_csv(fname, converters={'hex_col': partial(int, base=16)},
+                        dtype = {"hex_col" : "hex64"})
```
Comment (kkraus14) [inline python/cudf/cudf/utils/ioutils.py]: I think you need a `...` on this line similar to above multi-line statements

File: python/cudf/cudf/utils/ioutils.py [hunk 7]
```diff
@@ -772,6 +772,29 @@
 1  456 2018-11-14T12:35:01.000 5784
 2  789 2018-11-15T18:02:59.000 6117
 
+Read ``hexadecimal values`` from a csv file as integer column with cudf
```
Comment (randerzander) [inline python/cudf/cudf/utils/ioutils.py]: Can you also add a note or example about how `0x...` formatted numbers are parsed into ints if dtype inference is on?

**Comment by randerzander** [comment]
rerun tests

**Comment by kkraus14** [comment]
```

+ python -c 'import cudf'

Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/conda/conda-bld/cudf_1583851712040/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/cudf/__init__.py", line 4, in <module>

    from cudf import core, datasets

  File "/conda/conda-bld/cudf_1583851712040/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/cudf/core/__init__.py", line 2, in <module>

    from cudf.core import buffer, column

  File "/conda/conda-bld/cudf_1583851712040/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/cudf/core/column/__init__.py", line 1, in <module>

    from cudf.core.column.categorical import CategoricalColumn  # noqa: F401

  File "/conda/conda-bld/cudf_1583851712040/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/cudf/core/column/categorical.py", line 10, in <module>

    import cudf._lib as libcudf

  File "/conda/conda-bld/cudf_1583851712040/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/cudf/_lib/__init__.py", line 1, in <module>

    from . import (

  File "cudf/_lib/avro.pyx", line 18, in init cudf._lib.avro

  File "/conda/conda-bld/cudf_1583851712040/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/cudf/utils/ioutils.py", line 801, in <module>

    remote_data_sources=_docstring_remote_sources

KeyError: '"hex_col" '
```

**Comment by rnyak** [comment]
@kkraus14 @randerzander will take a look at that what's going on. If we want to merge other docs edits asap, I can take out 'hex_col` lines and then add again and create a new PR once I figure out why we get this error.

**Comment by kkraus14** [comment]
> @kkraus14 @randerzander will take a look at that what's going on. If we want to merge other docs edits asap, I can take out 'hex_col` lines and then add again and create a new PR once I figure out why we get this error.

Please do.

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**